pyiron_contrib/RDM/storagejob.py

```diff
@@ -0,0 +1,337 @@
+import os
+import posixpath
+import shutil
+
+from pyiron_base import GenericJob, DataContainer
+from pyiron_base.generic.filedata import FileData
+from pyiron_contrib.generic.s3io import FileS3IO
+
+
+def _remove_s3_working_directory(file_s3_io_handle, group_to_remove):
+    # Upon moving s3 to base, this should be moved to pyiron_base/job/util.py
+    if file_s3_io_handle.is_dir(group_to_remove):
+        file_s3_io_handle.remove_group(path=group_to_remove)
+
+
+class StorageType(DataContainer):
```

Comment:
I'm not convinced this needs to inherit from `DataContainer`.  This only stores a single string or am I missing more?  Since nothing else is stored in this I'd ask this to either manually implement `to_hdf`/`from_hdf` or use a `DataContainer` to store the `storage_type` as in the [example](https://pyiron-base.readthedocs.io/en/latest/apidoc/pyiron_base.generic.datacontainer.html?highlight=DataContainer).  This avoid the weird issues with `create_group` and using `*` in the `__init__`.

Reply:
I used the `DataContainer` since I like the features it has like the `.key` access to the fields, the already implemented `read_only`, and the easy creation of storage groups and their auto-hdf features I use in the `use_s3_storage().  However, right now it is probably not needed to store this in a `DataContainer` subclass at all since it would be possible to store it in a `job._store = DataContainer()` directly and just use a clever setter for the `_available_storage_types`.